src/openfermion/ops/_qubit_operator.py

```diff
@@ -105,67 +108,67 @@ def __imul__(self, multiplier):
             return self
 
         # Handle QubitOperator.
-        elif isinstance(multiplier, QubitOperator):
-            result_terms = dict()
-            for left_term in self.terms:
-                for right_term in multiplier.terms:
-                    new_coefficient = (self.terms[left_term] *
-                                       multiplier.terms[right_term])
-
-                    # Loop through local operators and create new sorted list
-                    # of representing the product local operator:
-                    product_operators = []
-                    left_operator_index = 0
-                    right_operator_index = 0
-                    n_operators_left = len(left_term)
-                    n_operators_right = len(right_term)
-                    while (left_operator_index < n_operators_left and
-                           right_operator_index < n_operators_right):
-                        (left_qubit, left_loc_op) = (
-                            left_term[left_operator_index])
-                        (right_qubit, right_loc_op) = (
-                            right_term[right_operator_index])
-
-                        # Multiply local operators acting on the same qubit
-                        if left_qubit == right_qubit:
-                            left_operator_index += 1
-                            right_operator_index += 1
-                            (scalar, loc_op) = _PAULI_OPERATOR_PRODUCTS[
-                                (left_loc_op, right_loc_op)]
-
-                            # Add new term.
-                            if loc_op != 'I':
-                                product_operators += [(left_qubit, loc_op)]
-                                new_coefficient *= scalar
-                            # Note if loc_op == 'I', then scalar == 1.0
-
-                        # If left_qubit > right_qubit, add right_loc_op; else,
-                        # add left_loc_op.
-                        elif left_qubit > right_qubit:
-                            product_operators += [(right_qubit, right_loc_op)]
-                            right_operator_index += 1
-                        else:
-                            product_operators += [(left_qubit, left_loc_op)]
-                            left_operator_index += 1
-
-                    # Finish the remaining operators:
-                    if left_operator_index == n_operators_left:
-                        product_operators += right_term[
-                            right_operator_index::]
-                    elif right_operator_index == n_operators_right:
-                        product_operators += left_term[left_operator_index::]
-
-                    # Add to result dict
-                    tmp_key = tuple(product_operators)
-                    if tmp_key in result_terms:
-                        result_terms[tmp_key] += new_coefficient
+        if not isinstance(multiplier, QubitOperator):
```

Comment:
Pylint showed too many branches, so reordered `if else` to reduce one branch since the previous `else` branch returns directly..
No change other than the reordering.